python/torch_mlir_e2e_test/test_suite/elementwise.py

```diff
@@ -1654,6 +1654,28 @@ def forward(self, x):
 def ElementwiseDivScalarModule_basic(module, tu: TestUtils):
     module.forward(tu.rand(3, 4))
 
+
+# ==============================================================================
+
+
+class ElementwiseDivIntScalarModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.int64, True),
+    ])
+    def forward(self, x):
+        return torch.div(x, 128)
+
+
+@register_test_case(module_factory=lambda: ElementwiseDivIntScalarModule())
+def ElementwiseDivIntScalarModule_basic(module, tu: TestUtils):
+    module.forward(tu.randint(3, 4))
```

Comment:
Was your original reproducer with an integer tensor as first argument? I though it was a float tensor divided by an integer scalar.
Here, I doubt that the result type is a floating point tensor.

Reply:
Yes it was the following line from `deepset_deberta-v3-base-squad2_AutoModelForQuestionAnswering_tosa_bf16` I believe:
```
torch.aten.div.Scalar %arg0, %int128 : !torch.vtensor<[?, ?],si64>, !torch.int  -> !torch.vtensor<[?, ?],f32>
```